### PR TITLE
feat(VTreeview): add selectionType prop

### PR DESCRIPTION
**Pull request by nekosaur**
## Description
Change default selection strategy, and include a new one.

## Motivation and Context
The current strategy of including parent in selection array when all its leafs are selected is causing some issues (see #6903)

## How Has This Been Tested?
playground, unit tests

## Markup:

<details>

```vue
<template>
  <v-layout>
    <v-flex>
      <v-card-text>
        <v-treeview
          v-model="tree"
          :load-children="fetch"
          :items="items"
          activatable
          active-class="grey lighten-4 indigo--text"
          selected-color="indigo"
          open-on-click
          selectable
          expand-icon="mdi-chevron-down"
          on-icon="mdi-bookmark"
          off-icon="mdi-bookmark-outline"
          indeterminate-icon="mdi-bookmark-minus"
          selection-type="independent"
          return-object
        >
        </v-treeview>
      </v-card-text>
    </v-flex>

    <v-divider vertical></v-divider>

    <v-flex
      xs12
      md6
    >
      <v-card-text>
        <v-chip
          v-for="item in tree"
          :key="item.name"
          color="grey"
          dark
          small
        >
          <v-icon left small>mdi-beer</v-icon>
          {{ item.name }}
          <v-icon @click="removeSelected(i)" >mdi-close</v-icon>
        </v-chip>
      </v-card-text>
    </v-flex>
  </v-layout>
</template>

<script>
  export default {
    data: () => ({
      breweries: [],
      isLoading: false,
      tree: [],
      types: []
    }),

    computed: {
      items () {
        const children = this.types.map(type => ({
          id: type,
          name: this.getName(type),
          children: this.getChildren(type)
        }))

        return [{
          id: 1,
          name: 'All Breweries',
          children
        }]
      },
      shouldShowTree () {
        return this.breweries.length > 0 && !this.isLoading
      }
    },

    watch: {
      breweries (val) {
        this.types = val.reduce((acc, cur) => {
          const type = cur.brewery_type

          if (!acc.includes(type)) acc.push(type)

          return acc
        }, []).sort()
      }
    },

    methods: {
      fetch () {
        if (this.breweries.length) return

        return fetch('https://api.openbrewerydb.org/breweries')
          .then(res => res.json())
          .then(data => (this.breweries = data))
          .catch(err => console.log(err))
      },
      getChildren (type) {
        const breweries = []

        for (const brewery of this.breweries) {
          if (brewery.brewery_type !== type) continue

          breweries.push({
            ...brewery,
            name: this.getName(brewery.name)
          })
        }

        return breweries.sort((a, b) => {
          return a.name > b.name ? 1 : -1
        })
      },
      getName (name) {
        return `${name.charAt(0).toUpperCase()}${name.slice(1)}`
      },
      removeSelected(i) {
        this.$delete(this.tree, i)
      },
    }
  }
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/data/pages/components/Treeview.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
           "type": "examples",
           "value": [
             { "file": "dense", "newIn": "2.0" },
+            { "file": "selectionType", "newIn": "2.0" },
```
Comment (jacekkarczmarczyk) [inline packages/docs/src/data/pages/components/Treeview.json]: selection-type

File: packages/docs/src/examples/treeview/selectionType.vue [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<template>
```
Comment (jacekkarczmarczyk) [inline packages/docs/src/examples/treeview/selectionType.vue]: selection-type.vue

File: packages/docs/src/lang/en/components/Treeview.json [hunk 3]
```diff
@@ -24,6 +24,10 @@
     "human-resources": {
       "header": "### Searching a directory",
       "desc": "Easily filter your treeview by using the **search** prop. You can easily apply your custom filtering function if you need case-sensitive or fuzzy filtering by setting the **filter** prop. This works similar to the [v-autocomplete](/components/autocompletes) component."
+    },
+    "selectionType": {
```
Comment (jacekkarczmarczyk) [inline packages/docs/src/lang/en/components/Treeview.json]: selection-type

File: packages/vuetify/src/components/VTreeview/VTreeview.ts [hunk 4]
```diff
@@ -80,6 +80,11 @@ export default mixins(
     } as PropValidator<NodeArray>,
     search: String,
     filter: Function as PropValidator<FilterTreeItemFunction>,
+    selectionType: {
+      type: String,
+      default: 'leaf',
+      validator: (v: string) => ['leaf', 'independent'].includes(v),
+    },
```
Comment (sh7dm) [inline packages/vuetify/src/components/VTreeview/VTreeview.ts]:
```suggestion
    } as PropValidator<'leaf' | 'independent'>,
```

**Pull request merged**